File: PieGallery/app/src/main/java/com/deadely/piegallery/di/module/DatabaseModule.java
package com.deadely.piegallery.di.module;

import com.deadely.piegallery.App;
import com.deadely.piegallery.database.AppDBHelper;
import com.deadely.piegallery.database.dao.FavoritesDao;
import com.deadely.piegallery.database.dao.PhotoDao;

import dagger.Module;
import dagger.Provides;

@Module
public class DatabaseModule {

    @Provides
    AppDBHelper getDBIntance() {
        return App.getInstance().getDatabaseInstance();
    }

    @Provides
    FavoritesDao getFavDAO() {
        return App.getInstance().getDatabaseInstance().getFavoritesDao();
    }

    @Provides
    PhotoDao getPhotoDAO() {
        return App.getInstance().getDatabaseInstance().getPhotoDao();
    }
}

File: itlen/app/src/main/java/com/deadely/itl_en/di/module/DatabaseModule.kt
package com.deadely.itl_en.di.module

import com.deadely.itl_en.App.Companion.instance
import com.deadely.itl_en.database.AppDatabase
import dagger.Module
import dagger.Provides

@Module
class DatabaseModule {
    @Provides
    fun providedBInstance(): AppDatabase {
        return instance.db
    }
}